### PR TITLE
Add HTTP API reference

### PR DESCRIPTION
**Pull request by justinweng-instaclustr**
Add documentation for some HTTP API endpoints added in https://github.com/uber/cadence/pull/5338 and https://github.com/uber/cadence/pull/5305.

### CODE REVIEW

File: src/docs/03-concepts/10-http-api.md [hunk 1]
```diff
@@ -5,208 +5,1201 @@ permalink: /docs/concepts/http-api
...
+  },
+  "taskList": {
+    "name": "tasklist-name"
+  },
+  "identity": "My custom caller identity",
+  "requestId": "4D1E4058-6FCF-4BA8-BF16-8FA8B02F9651"
+} 
+```
+
+Describe a cluster using curl command
```
Comment (ronycsdu) [inline src/docs/03-concepts/10-http-api.md]: shouldn't this be under Admin API?

File: src/docs/03-concepts/10-http-api.md [hunk 2]
```diff
@@ -5,208 +5,1201 @@ permalink: /docs/concepts/http-api
...
+  ],
+  "nextPageToken": ""
+}
+```
+
+</details>
+
+------------------------------------------------------------------------------------------
+
+### Meta API
```
Comment (ronycsdu) [inline src/docs/03-concepts/10-http-api.md:881]: If you are publishing this without any content initially, please write:
    **Coming Soon...**

File: src/docs/03-concepts/10-http-api.md [hunk 3]
```diff
@@ -5,208 +5,1201 @@ permalink: /docs/concepts/http-api
...
+
+------------------------------------------------------------------------------------------
+
+### Visibility API
```
Comment (ronycsdu) [inline src/docs/03-concepts/10-http-api.md]: If you are publishing this without any content initially, please write:
    **Coming Soon...**

File: src/docs/03-concepts/10-http-api.md (hunk 1 above)
Comment (justinweng-instaclustr) [reply src/docs/03-concepts/10-http-api.md]: It's also in Admin API section. This part only serves as an initial example. Do we want to remove it here and only show DescribeCluster in the Admin API section?

File: src/docs/03-concepts/10-http-api.md (hunk 2 above)
Comment (justinweng-instaclustr) [reply src/docs/03-concepts/10-http-api.md:881]: Yeah I'll add more examples and will write `Coming Soon` in empty sections.

File: src/docs/03-concepts/10-http-api.md [hunk 4]
```diff
@@ -5,208 +5,2180 @@ permalink: /docs/concepts/http-api
 ---
 
 # Using HTTP API
```
Comment (ronycsdu) [inline src/docs/03-concepts/10-http-api.md:7]: Since Cadence docs are not versioned, we should mention that this only applies to Cadence v1.2.1+

Comment (justinweng-instaclustr) [reply src/docs/03-concepts/10-http-api.md:7]: Added this to the introduction and rephrased the introduction using GPT4

File: src/docs/03-concepts/10-http-api.md [hunk 5]
```diff
@@ -5,208 +5,2181 @@ permalink: /docs/concepts/http-api
...
-Cadence allows you to interact with server using HTTP API. What does it mean?  HTTP/JSON communication is a versatile way to interact with servers. For Cadence, it means you can expose a list of RPC methods to be invoked using HTTP protocol. This opens possibilities to interact with Cadence server using any programming language with HTTP support. This can be used to start or terminate workflows from your bash scripts, observe cluster status or invoke any other operation supported by Cadence RPC declaration. 
 
- 
+## Introduction
 
-## Setup 
+From version 1.2.1 onwards, Cadence has introduced HTTP API support, which allows you to interact with the Cadence server
```
Comment (ronycsdu) [inline src/docs/03-concepts/10-http-api.md]: maybe you want emphasize (bold) the phrase "version 1.2.1"?

Comment (justinweng-instaclustr) [reply src/docs/03-concepts/10-http-api.md]: Done

**Comment by justinweng-instaclustr** [comment]
New PR opened for https://github.com/uber/Cadence-Docs is here https://github.com/uber/Cadence-Docs/pull/164. Close this PR.

**Pull request closed without merging**